storage/preferences/src/main/java/de/danoeh/antennapod/storage/preferences/UserPreferences.java

```diff
@@ -546,6 +553,39 @@ public static int getRewindSecs() {
         return prefs.getInt(PREF_REWIND_SECS, 10);
     }
 
+    public static String[] getAutodownloadSelectedNetworks() {
+        String selectedNetWorks = prefs.getString(PREF_AUTODL_SELECTED_NETWORKS, "");
+        return TextUtils.split(selectedNetWorks, ",");
+    }
+
+    public static boolean areNetworkConstraintsDisabled() {
+        String untilPref = prefs.getString(PREF_NETWORK_CONSTRAINTS_DISABLED_UNTIL, LocalDateTime.now().minusMinutes(1).toString());
+        if (untilPref == null) { return false; }
+        LocalDateTime until = LocalDateTime.parse(untilPref);
+        LocalDateTime now = LocalDateTime.now();
+        return now.isBefore(until);
+    }
+
+    public static void setPrefNetworkConstraintsDisabledTimespanMinutes(int timespanMinutes) {
+        if (timespanMinutes < 0) { timespanMinutes = 0; }
+        prefs.edit().putInt(PREF_NETWORK_CONSTRAINTS_DISABLED_TIMESPAN_MINUTES,timespanMinutes).apply();
+    }
+
+    public static void setPrefNetworkConstraintsDisabled(boolean isEnabled) {
+        int minutes = prefs.getInt(PREF_NETWORK_CONSTRAINTS_DISABLED_TIMESPAN_MINUTES,10);
+        LocalDateTime until = (isEnabled) ? LocalDateTime.now().plus(minutes,ChronoUnit.MINUTES) : LocalDateTime.now().minusMinutes(1);
```

Comment:
I don't think we need to store this in the preferences. Just a static variable in DownloadActionButton should be enough, because it is not persistent data